src/app.rs

```diff
@@ -815,13 +821,86 @@ impl Window {
         });
     }
 
-    fn controls(&mut self, ui: &mut egui::Ui, cx: &Context) {
+    fn modify_interval(&mut self, ui: &mut egui::Ui, cx: &mut Context) {
+        ui.subheading("Interval", cx);
+        let start_res = ui
+            .horizontal(|ui| {
+                ui.label("Start:");
+                ui.text_edit_singleline(&mut cx.view_interval_start_buffer)
+            })
+            .inner;
+
+        let stop_res = ui
+            .horizontal(|ui| {
+                ui.label("Stop:");
+                ui.text_edit_singleline(&mut cx.view_interval_stop_buffer)
+            })
+            .inner;
+
+        if start_res.lost_focus()
+            && cx.view_interval_start_buffer != cx.view_interval.start.to_string()
+        {
+            match Interval::convert_str_to_timestamp(&cx.view_interval_start_buffer) {
+                Ok(start) => {
+                    // validate timestamp
+                    if start > cx.view_interval.stop {
+                        cx.view_interval_start_buffer = "Start must be before stop".to_string();
```

Comment:
I was wondering if, instead of replacing the contents of the interval string, we should pop up an error message right below this widget. This has the advantage of not throwing away the user's entry if they typo'd something.